formats/protobuf/src/test/java/io/cloudevents/protobuf/ProtobufFormatTest.java

```diff
@@ -0,0 +1,180 @@
+
+        // Load the expected representation
+        Message expectedProto = loadProto(jsonFile);
+
+        // Compare.
+        assertThat(actualProto).ignoringRepeatedFieldOrder().isEqualTo(expectedProto);
+
+    }
+
+    /**
+     * RoundTrip Test
+     * <p>
+     * Steps:
+     * (1) Load the raw proto representation.
+     * (2) Deserialize into a CE using the ProtoFormat
+     * (3) Serialize the CE into a buffer using the ProtoFormat
+     * (4) re-hydrate the protobuf that was serialized in step (3)
+     * (5) Ensure the proto from (1) equals the proto from (4)
+     *
+     * @param filename
+     * @throws IOException
+     */
+    @ParameterizedTest
+    @MethodSource("roundTripTestArguments")
+    public void roundTripTest(String filename) throws IOException {
+
+        // Load the source (expected) raw proto wire represention.
+        byte[] rawData = getProtoData(filename);
+
+        // Create the CloudEvent
+        io.cloudevents.CloudEvent expEvent = format.deserialize(rawData);
+
+        // Sanity
+        assertNotNull(expEvent);
+
+        // Serialise it back out.
+        byte[] raw = format.serialize(expEvent);
+
+        // Sanity
+        assertNotNull(raw);
+        assertTrue(raw.length > 0);
+
+        // Now read it back
+        CloudEvent newProto = CloudEvent.parseFrom(raw);
+
+        // Now hopefully these will match
+
+        CloudEvent expectedProto = CloudEvent.parseFrom(rawData);
+        assertThat(newProto).ignoringRepeatedFieldOrder().isEqualTo(expectedProto);
+
+    }
+
+    public static Stream<Arguments> serializeTestArgumentsDefault() {
+        return Stream.of(
+            Arguments.of(V1_MIN, "v1/min.proto.json"),
+            Arguments.of(V1_WITH_JSON_DATA, "v1/json_data.proto.json"),
+            Arguments.of(V1_WITH_TEXT_DATA, "v1/text_data.proto.json"),
+            Arguments.of(V1_WITH_JSON_DATA_WITH_EXT, "v1/json_data_with_ext.proto.json"),
+            Arguments.of(V1_WITH_XML_DATA, "v1/xml_data.proto.json"),
+            Arguments.of(V1_WITH_BINARY_EXT, "v1/binary_ext.proto.json"),
+
+            Arguments.of(V03_MIN, "v03/min.proto.json")
+
+        );
+    }
+
+    public static Stream<String> roundTripTestArguments() {
+        return Stream.of(
+
+            "v1/min.proto.json",
+            "v1/json_data.proto.json",
+            "v1/text_data.proto.json",
+            "v1/json_data_with_ext.proto.json",
+            "v1/xml_data.proto.json",
+            "v1/binary_ext.proto.json",
+
+            "v03/min.proto.json"
+        );
+    }
+
+
+    // ----------------------------------------------------------------
+
+    private static Message loadProto(String filename) {
+        CloudEvent.Builder b = CloudEvent.newBuilder();
+        Message retVal = null;
+        try {
+
+            JsonFormat.parser().ignoringUnknownFields().merge(getReader(filename), b);
+            retVal = b.build();
+
+        } catch (IOException e) {
+            fail(e);
```

Comment:
Just change the signature to `throws IOException` and every test `throws Exception`

Reply:
ack